best_practices/security.rst

```diff
@@ -173,6 +182,25 @@ the same ``getAuthorEmail`` logic you used above:
 
             return false;
         }
+
+        /**
+         * Checks if the user token has the given role taking into account the
+         * entire role hierarchy if defined by the application.
+         */
+        private function userHasRole(UserInterface $user, $roleName)
+        {
+            if (null === $this->roleHierarchy) {
+                return in_array($roleName, $user->getRoles(), true);
+            }
+
+            foreach ($this->roleHierarchy->getReachableRoles($user->getRoles()) as $role) {
```

Comment:
``$user->getRoles()`` can represent roles as strings, which is not supported by ``getReachableRoles`` (the Token class is normalizing this, and ``getReachableRoles`` is meant to work with ``$token->getRoles()``